src/gui/guiFormSpecMenu.cpp

```diff
@@ -2576,6 +2576,41 @@ void GUIFormSpecMenu::parseAnchor(parserData *data, const std::string &element)
 			<< "'" << std::endl;
 }
 
+bool GUIFormSpecMenu::parsePaddingDirect(parserData *data, const std::string &element)
+{
+	if (element.empty())
+		return false;
+
+	std::vector<std::string> parts = split(element, '[');
+
+	if (parts.size() != 2)
```

Comment:
This and `anchor[]` really need an additional check for `FORMSPEC_API_VERSION` to allow backwards compatibility in the future (= forwards compatibility ?).

```C++
if (parts.size() < 2 || (parts.size() > 2 && m_formspec_version <= FORMSPEC_API_VERSION))
	return false;
```
or after De Morgan's Law:
```
if (!(parts.size() == 2 || (parts.size() > 2 && m_formspec_version > FORMSPEC_API_VERSION)))
	return false;
```

Reply:
Heh, not here, unless you're suggesting future compatibility with elements like `anchor[stuff[stuff]` :P  But yeah, needs to be added to `parsePadding` and friends.

Reply:
Oops. Yes exactly. I even made a mistake at suggesting this for Comma-separated arguments. This compatibility code would allow for `a,b,c,d` arguments but not new ones which are separated by semicolons. As of how it currently is, it sadly does not make much sense. What would you think about splitting by semicolons first, then by commas for the first argument parsing?

Reply:
Dang, that's true.  I'm tempted to say "I don't care", but it's probably good to get right.